src/crs_linter/linter.py

```diff
@@ -955,3 +958,37 @@ def check_capture_action(self):
                         captured_var_chain_level = 0
                         use_captured_var = False
                         ruleid = 0
+
+    def find_ids_without_tests(self, test_cases, exclusion_list):
+        """
+            s: the parsed structure
+            test_cases: all available test cases
+            exclusion_list_name: file which contains rule id's without tests
+        """
+        rids = {}
+        for i in self.data:
+            # only SecRule counts
+            if i['type'] == "SecRule":
+                for a in i['actions']:
+                    # find the `id` action
+                    if a['act_name'] == "id":
+                        # get the argument of the action
+                        rid = int(a['act_arg']) # int
+                        srid = a['act_arg']     # string
+                        if (rid%1000) >= 100:   # skip the PL control rules
+                            # also skip these hardcoded rules
+                            need_check = True
+                            for excl in exclusion_list:
```

Comment:
If you make `exclusion_list` a `dict` instead, you can just perform a lookup. That's also much more performant.

Reply:
Sure, I can do that later too.

Meanwhile I realized that the linter has an own file reader function, so I changed the read method by that here. That function produces a list, and we use that (now) at three times. This is why I ask to fix this later.